torch/distributions/truncated_distribution.py

```diff
@@ -0,0 +1,61 @@
+import torch
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+class TruncatedDistribution(Distribution):
+    r"""
+    Extension of the Distribution class, which applies truncation to base distribution.
+    """
+    def __init__(self, base_distribution, lower_bound=-float('inf'), upper_bound=float('inf'), *args, **kwargs):
+        super(TruncatedDistribution, self).__init__(*args, **kwargs)
+        self.base_dist = base_distribution
+        self.lower_bound, self.upper_bound, _ = broadcast_all(lower_bound, upper_bound,
+                                                              getattr(self.base_dist,
```

Comment:
This looks really dangerous. Why do we need to broadcast? Can we simply set
```py
self.lower_bound = lower_bound
self.upper_bound = upper_bound
```

Reply:
I was thinking about supporting batched bounds while writing that part, but I gave up that idea later & forgot to change it.